### PR TITLE
BHBC-620: Cleanup/refactor

### PR DESCRIPTION
**Pull request by NickPhura**
<details open>
  <summary><a href="https://quartech.atlassian.net/browse/BHBC-620" title="BHBC-620" target="_blank">BHBC-620</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>Endpoint and code refactoring to support Project editing</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Story
        Story
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Progress</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td><a href="https://quartech.atlassian.net/issues?jql=project%20%3D%20BHBC%20AND%20labels%20%3D%20API%20ORDER%20BY%20created%20DESC" title="API">API</a></td>
    </tr>
  </table>
</details>

---

# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-620

## This PR contains the following changes

Lots of refactoring, mainly around file/function naming, and some folder structure

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [x] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

**Comment by anissa-agahchen** [comment]
I'm getting a CORS error on create:
Screen Shot 2021-03-22 at 12 50 18 PM

File: api/src/openapi/schemas/project.ts [hunk 1]
```diff
@@ -174,9 +174,375 @@ export const projectPostBody = {
...
+              },
+              sampling_conducted: {
+                type: 'string'
+              }
+            }
+          }
+        }
+      }
+    },
+    project: {
```
Comment (sdevalapurkar) [inline api/src/openapi/schemas/project.ts]: might need to add iucn classification to this

File: api/src/openapi/schemas/project.ts [hunk 2]
```diff
@@ -174,9 +174,375 @@ export const projectPostBody = {
...
```
Comment (sdevalapurkar) [inline api/src/openapi/schemas/project.ts]: same here

File: api/src/paths/project/{projectId}/update.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+PUT.apiDoc = {
+  description: 'Update a project.',
+  tags: ['project'],
+  security: [
+    {
+      Bearer: WRITE_ROLES
+    }
+  ],
+  requestBody: {
+    description: 'Project post request object.',
```
Comment (sdevalapurkar) [inline api/src/paths/project/{projectId}/update.ts]:
```suggestion
    description: 'Project put request object.',
```

File: api/src/paths/project/{projectId}/update.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      $ref: '#/components/responses/503'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+/**
+ * Update a project.
```
Comment (sdevalapurkar) [inline api/src/paths/project/{projectId}/update.ts]: seems like a copy paste error with the comment here perhaps

**Comment by NickPhura** [comment]
> 
> 
> I'm getting a CORS error on create:
> Screen Shot 2021-03-22 at 12 50 18 PM

Forgot to update the url in the app api.  Pushed a commit to fix it

**Review by sdevalapurkar — COMMENTED** [review]
this will need some changes after https://github.com/bcgov/biohubbc/pull/155 goes in
but otherwise looking good

**Review by sdevalapurkar — COMMENTED** [review]
looks good when merge conflicts are resolved

**Review by sdevalapurkar — APPROVED** [review]
LGTM 🥇 once CI passes

**Pull request merged**